ArduCopter/Copter.cpp

```diff
@@ -273,32 +273,37 @@ void Copter::get_scheduler_tasks(const AP_Scheduler::Task *&tasks,
 
 constexpr int8_t Copter::_failsafe_priorities[7];
 
-#if AP_SCRIPTING_ENABLED
+
+#if AP_SCRIPTING_ENABLED || AP_EXTERNAL_CONTROL_ENABLED
 #if MODE_GUIDED_ENABLED == ENABLED
-// start takeoff to given altitude (for use by scripting)
-bool Copter::start_takeoff(float alt)
+// set target location (for use by external control and scripting)
+bool Copter::set_target_location(const Location& target_loc)
 {
     // exit if vehicle is not in Guided mode or Auto-Guided mode
     if (!flightmode->in_guided_mode()) {
         return false;
     }
 
-    if (mode_guided.do_user_takeoff_start(alt * 100.0f)) {
-        copter.set_auto_armed(true);
-        return true;
-    }
-    return false;
+    return mode_guided.set_destination(target_loc);
 }
+#endif //MODE_GUIDED_ENABLED == ENABLED
+#endif //AP_SCRIPTING_ENABLED || AP_EXTERNAL_CONTROL_ENABLED
 
-// set target location (for use by scripting)
-bool Copter::set_target_location(const Location& target_loc)
+#if AP_SCRIPTING_ENABLED
+#if MODE_GUIDED_ENABLED == ENABLED
+// start takeoff to given altitude (for use by scripting)
+bool Copter::start_takeoff(float alt)
 {
     // exit if vehicle is not in Guided mode or Auto-Guided mode
     if (!flightmode->in_guided_mode()) {
         return false;
     }
 
-    return mode_guided.set_destination(target_loc);
+    if (mode_guided.do_user_takeoff_start(alt * 100.0f)) {
```

Comment:
This looks like it enables taking off from guided mode using a regular command.  That's a bit of a safety issue so we should do this.